homeassistant/components/netgear/device_tracker.py

```diff
@@ -39,132 +42,158 @@
 
-        Returns boolean if scanning successful.
-        """
-        _LOGGER.debug("Scanning")
 
-        results = self.get_attached_devices()
+@callback
+def add_entities(router, async_add_entities, tracked):
+    """Add new tracker entities from the router."""
+    new_tracked = []
 
-        if _LOGGER.isEnabledFor(logging.DEBUG):
-            _LOGGER.debug("Scan result: \n%s", pformat(results))
+    for mac, device in router.devices.items():
+        if mac in tracked:
+            continue
 
-        if results is None:
-            _LOGGER.warning("Error scanning devices")
+        new_tracked.append(NetgearDeviceEntity(router, device))
+        tracked.add(mac)
 
-        self.last_results = results or []
+    if new_tracked:
+        async_add_entities(new_tracked, True)
 
-    def get_attached_devices(self):
-        """List attached devices with pynetgear.
 
-        The v2 method takes more time and is more heavy on the router
-        so we only use it if we need connected AP info.
-        """
-        if self.tracked_accesspoints:
-            return self._api.get_attached_devices_2()
+class NetgearDeviceEntity(ScannerEntity):
+    """Representation of a device connected to a Netgear router."""
 
-        return self._api.get_attached_devices()
+    def __init__(self, router: NetgearRouter, device) -> None:
+        """Initialize a Netgear device."""
+        self._router = router
+        self._device = device
+        self._mac = device["mac"]
+        self._name = self.get_device_name(device)
+        self._icon = DEVICE_ICONS.get(device["device_type"], "mdi:help-network")
+        self._active = device["active"]
+        self._attrs = {}
+
+    def get_device_name(self, device):
+        """Return the name of the given device or the MAC if we don't know."""
+        name = device["name"]
+        if not name or name == "--":
+            name = self._mac
+
+        return name
+
+    @callback
+    def update_device(self) -> None:
+        """Update the Netgear device."""
+        self._device = self._router.devices[self._mac]
+        self._active = self._device["active"]
+        self._icon = DEVICE_ICONS.get(self._device["device_type"], "mdi:help-network")
+        self._attrs = {
+            "link_type": self._device["type"],
+            "link_rate": self._device["link_rate"],
+            "signal_strength": self._device["signal"],
+        }
+        if not self._active:
+            self._attrs = {}
+
+        self.async_write_ha_state()
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._mac
+
+    @property
+    def name(self) -> str:
+        """Return the name."""
+        return self._name
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the router."""
+        return self._active
+
+    @property
+    def source_type(self) -> str:
+        """Return the source type."""
+        return SOURCE_TYPE_ROUTER
+
```

Comment:
Do we have the hostname?

If so we can set the `hostname` property

Reply:
not the hostname but we do have the IP i think, testing now.
Is the IP adress also acceptable?

Reply:
It must be the `hostname`.  Some routers call this the `name`

Reply:
https://developers.home-assistant.io/docs/core/entity/device-tracker/#dhcp-discovery

Reply:
I think the name property is indeed the hostname, at least it is correct for my windows PC.
How can I easily find the hostname of a device (lets say a shelly 1 for example)?
Then I can compare if they are the same.

Reply:
You'd have to watch the dhcp packets.  Example in the `dhcp` integration

Reply:
Turn on debug logging for `homeassistant.components.dhcp` and restart the device

Reply:
Also unless manually, I don't think there is any other place it would be likely to get the name so `name` is almost definitely the hostname in that case

Reply:
I think it is indeed the hostname as far as I can see, I added the propery